docs/fsharp/language-reference/byrefs.md

```diff
@@ -0,0 +1,243 @@
+
+All of these rules together mean that the holder of an `inref` pointer may not modify the immediate contents of the memory being pointed to.
+
+### Outref semantics
+
+The purpose of `outref<'T>` is for documenting that the pointer should only be read from. Unexpectedly, `outref<'T>` permits reading the underlying value despite its name. This is for compatibility purposes. Semantically, `outref<'T>` is no different than `byref<'T>`.
+
+### Interop with C #
+
+C# supports the `in ref` and `out ref` keywords, in addition to `ref` returns. The following table shows how F# interprets what C# emits:
+
+|C# construct|F# infers|
+|------------|---------|
+|`ref` return value|`outref<'T>`|
+|`ref readonly` return value|`inref<'T>`|
+|`in ref` parameter|`inref<'T>`|
+|`out ref` parameter|`outref<'T>`|
+
+The following table shows what F# emits:
+
+|F# construct|Emitted construct|
+|------------|-----------------|
+|`inref<'T>` argument|`[In]` attribute on argument|
+|`inref<'T>` return|`modreq` attribute on value|
+|`inref<'T>` in abstract slot or implementation|`modreq` on argument or return|
+|`outref<'T>` argument|`[Out]` attribute on argument|
+
+### Type inference and overloading rules
+
+An `inref<'T>` type is inferred by the F# compiler in the following cases:
+
+1. A .NET parameter or return type that has an `IsReadOnly` attribute.
+2. The `this` pointer on a struct type that has no mutable fields.
+3. The address of a memory location derived from another `inref<_>` pointer.
+
+When an implicit address of an `inref` is being taken, an overload with an argument of type `SomeType` is preferred to an overload with an argument of type `inref<SomeType>`. For example:
+
+```fsharp
+type C() =
+    static member M(x: System.DateTime) = x.AddDays(1.0)
+    static member M(x: inref<System.DateTime>) = x.AddDays(2.0)
+    static member M2(x: System.DateTime, y: int) = x.AddDays(1.0)
+    static member M2(x: inref<System.DateTime>, y: int) = x.AddDays(2.0)
+
+let res = System.DateTime.Now
+let v =  C.M(res)
+let v2 =  C.M2(res, 4)
+```
+
+In both cases, the overloads taking `System.DateTime` are resolved rater than the overloads taking `inref<System.DateTime>`.
+
+## Byref-like structs
+
+In addition to the `byref`/`inref`/`outref` trio, you can define your own structs that can adhere to `byref`-like semantics. This is done with the <xref:System.Runtime.CompilerServices.IsByRefLikeAttribute> attribute:
+
+```fsharp
+open System
+open System.Runtime.CompilerServices
+
+[<IsByRefLike; Struct>]
+type S(count1: Span<int>, count2: Span<int>) =
+    member x.Count1 = count1
+    member x.Count2 = count2
+```
+
+`IsByRefLike` does not imply `Struct`. Both must be present on the type.
+
+A "`byref`-like" struct in F# is a stack-bound value type. It is never allocated on the managed heap. They are useful for high-performance programming, as they are enforced with set of strong checks about the lifetimes and non-capture. These rules are:
```

Comment:
There's a change here from singular (it = a byref-like struct) to plural (they = byref-like structs). This should be consistent, preferably singular. That's also true of the following list of rules.